### ISSUE TITLE
What does gamma (margin) mean in your hyper parameter?

### ISSUE THREAD

**Issue by ZifengDing** [malllabiisc/CompGCN#17]
Hello,
Thanks for the great work. I am taking your work as my baseline.
I notice that you have a parameter called gamma, which is "margin" in your description. 
As I understand, you did not include learning rate decay in your code. In the function "fit" of "run.py", what you did is just to stop training after continuous 25 epochs without improving MRR. So what exactly does this "margin" mean here? 

Best Regards,
Zifeng

**Comment by soumyasanyal** [comment]
Hi Zifeng,

The margin `gamma` is a specific parameter of the `CompGCN_TransE` model (check [this line](https://github.com/malllabiisc/CompGCN/blob/3b06f5fec42526faa81afc158df9e64a8382982c/model/models.py#L65)). You can find more details about the margin-based loss in the original TransE [paper](https://papers.nips.cc/paper/5071-translating-embeddings-for-modeling-multi-relational-data.pdf). We've also discussed the need to decrease `gamma` in [this issue](https://github.com/malllabiisc/CompGCN/issues/9#issuecomment-597660674).

**Comment by ZifengDing** [comment]
Cool, now I get it. I was not taking TransE as the score function, so I missed this part.
Thanks a lot for your quick response and help!
